src/audio/tdfb/tdfb.c

```diff
@@ -356,6 +590,72 @@ static int tdfb_cmd_set_data(struct comp_dev *dev,
 	return ret;
 }
 
+static int tdfb_cmd_enum_set(struct sof_ipc_ctrl_data *cdata, struct tdfb_comp_data *cd)
+{
+	if (cdata->num_elems != 1)
+		return -EINVAL;
+
+	if (cdata->chanv[0].value > SOF_TDFB_MAX_ANGLES)
+		return -EINVAL;
+
+	switch (cdata->index) {
+	case CTRL_INDEX_AZIMUTH:
+		cd->az_value = cdata->chanv[0].value;
+		cd->update = true;
+		break;
+	case CTRL_INDEX_AZIMUTH_ESTIMATE:
+		cd->az_value_estimate = cdata->chanv[0].value;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int tdfb_cmd_switch_set(struct sof_ipc_ctrl_data *cdata, struct tdfb_comp_data *cd)
+{
+	if (cdata->num_elems != 1)
+		return -EINVAL;
+
+	switch (cdata->index) {
+	case CTRL_INDEX_PROCESS:
+		cd->beam_on = cdata->chanv[0].value;
+		cd->update = true;
+		break;
+	case CTRL_INDEX_DIRECTION:
+		cd->direction_updates = cdata->chanv[0].value;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int tdfb_cmd_set_value(struct comp_dev *dev, struct sof_ipc_ctrl_data *cdata)
+{
+	struct tdfb_comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
```

Comment:
ditto with initialisation and `return` from each case below

Reply:
done